src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -246,7 +246,11 @@ MergeTreeData::MergeTreeData(
 
     /// Check sanity of MergeTreeSettings. Only when table is created.
     if (!attach)
+    {
+        const auto & changes = metadata_.settings_changes->as<const ASTSetQuery &>().changes;
+        getContext()->checkMergeTreeSettingsConstraints(getContext()->getMergeTreeSettings(), changes);
```

Comment:
It should be correct to call `getReplicatedMergeTreeSettings()` for replicated engines here instead. But I'm not sure about the best way to propagate information about whether the table is replicated. Maybe better to do this check in subclasses?

Reply:
maybe do this check in `registerStorageMergeTree` `create` function?